usr/sbin/pkcsslotd/mutex.c

```diff
@@ -28,32 +28,37 @@ int CreateXProcLock(void)
     mode_t mode = (S_IRUSR | S_IRGRP);
 
     if (xplfd == -1)
-        xplfd = open(OCK_API_LOCK_FILE, O_RDONLY);
+        xplfd = open(OCK_API_LOCK_FILE, O_WRONLY);
 
     if (xplfd == -1) {
-        xplfd = open(OCK_API_LOCK_FILE, O_CREAT | O_RDONLY, mode);
+        /*
+        * Linux implementation of flock doesn't care about the mode the file
+        * is opened in, but AIX does - to obtain exclusive locks, AIX needs
+        * the file to be opened in write mode. Minority wins.
+        */
+        xplfd = open(OCK_API_LOCK_FILE, O_CREAT | O_WRONLY, mode);
 
         if (xplfd != -1) {
             if (fchmod(xplfd, mode) == -1) {
                 DbgLog(DL0, "%s:fchmod(%s):%s\n",
-                       __func__, OCK_API_LOCK_FILE, strerror(errno));
+                        __func__, OCK_API_LOCK_FILE, strerror(errno));
```

Comment:
why these changes? This makes the indention off by 1 character. same for the next few hunks